Add PR template (#272)
Financial-Times/dotcom-tool-kit: pull_request_template.md

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 

# Checklist:

> - [ ] My branch has been rebased onto the latest main (don't merge in main to your branch)

Reviewer: **nitpick**: the git terminology could be made a little clearer
```suggestion
- [ ] My branch has been rebased onto the latest commit on main (don't merge main into your branch)
```
